go-controller/pkg/allocator/pod/pod_annotation_test.go

```diff
@@ -575,6 +580,130 @@ func Test_allocatePodAnnotationWithRollback(t *testing.T) {
 			wantReleasedIPsOnRollback: ovntest.MustParseIPNets("192.168.0.3/24"),
 			role:                      types.NetworkRolePrimary, // has to be primary network for default routes to be set
 		},
+
+		{
+			// on networks with IPAM, honor a IP and MAC request through the network
+			// selection element
+			name:                            "expect requested MAC and IP, IPAM",
+			ipam:                            true,
+			enablePreconfiguredUDNAddresses: true,
+			role:                            types.NetworkRolePrimary, // has to be primary network for default routes to be set
+			netConf: &ovncnitypes.NetConf{
+				Topology: types.Layer2Topology,
+				NetConf: cnitypes.NetConf{
+					Name: "network",
+				},
+				NADName: util.GetNADName("namespace", "network"),
+				Subnets: "192.168.0.0/24,2001:db8::/64",
+				Role:    types.NetworkRolePrimary,
```

Comment:
this is fishy. We're passing the role attribute directly, but that is part of the `NetConf` struct. 

At the very least - IMHO - we can only pass this attribute via one way.

Reply:
@maiqueb I am going to remove the "netConf" option totally from the tests, I was just missing the allowPersistentIPs flag on the test
to make it work

Reply:
that works for me - we either pass netconf w/ everything in it, *or* we pass the individual knobs and compose a netconf in the test. 

Doing both makes this even harder to follow.